docs/source/architecture/sub_stakes.rst

```diff
@@ -0,0 +1,278 @@
+Sub-stakes
+==========
+
+A staker may extend the unlock period for any number of portions of their total stake. This divides the stake into smaller parts, each with a unique unlock date in the future. Stakers may also acquire and lock new tokens. The total stake is represented as the sum of all the different sub-stakes active in a given period (new period every 24h), which includes locked sub-stakes, and any sub-stakes that have passed their unlock date, and can be freely withdrawn. 
+
+Each sub-stake has a beginning and locking duration. For our terminology, we define `locking duration` to be the required number of times that `confirmActivity()` needs to be called so that the sub-stake is registered as locked for the subsequent period. For example, if a sub-stake has a *locking duration of 5 periods* it means that the sub-stake will be locked for periods: 1,2,3,4,5 because `confirmActivity()` would be called during periods 0,1,2,3,4 each time confirming activity for the subsequent period (1,2,3,4,5) respectively. Unless the starting period is specified, the sub-stake is considered locked for the 0th (current) period by default. When a staker confirms activity each day and winding down is enabled, the remaining lock time for relevant sub-stakes is reduced.
+
+A sub-stake remains active until it becomes unlocked, and a staker gets the reward for the last period by calling ``mint()`` or ``confirmActivity()`` once the last period is surpassed. Each staker can have no more than 30 active sub-stakes which are stored in an array. All sub-stake changes initially reuse slots of inactive sub-stakes for storage in the array, and if there are none, will instead use empty slots. Therefore, attempting to retrieve data about previous inactive sub-stakes is not guaranteed to be successful since the data could have been overwritten.
+
+
+
+Operations that modify the sub-stake array
+------------------------------------------
+
+Deposit and locking
+^^^^^^^^^^^^^^^^^^^
+*Methods* : ``deposit(uint256,uint16)``,  ``deposit(address,uint256,uint16)``,  ``lock(uint256,uint16)``
+
+To become a staker, NU tokens must be transferred to the ``StakingEscrow`` contract and locked using one of the ``deposit()`` methods. If the staker already has unlocked tokens within the account in the contract, then the stake can be locked using the ``lock()`` method. If successful, a new element will be created in the array of sub-stakes with the next period as the starting date, and the duration equal to the input parameter.
+
+
+**Example:**
+
+    A staker deposits 900 tokens:
+        * 1st sub-stake = 900 tokens starting from the next period and locked for 5 periods
```

Comment:
we should use our 'locking duration' term - similar comment for other spots on the page.
```suggestion
        * 1st sub-stake = 900 tokens starting from the next period and a locking duration of 5 periods
```

Reply:
✔️